### PR TITLE
:bug: fixing evaluation error due to changing NoiseMaskLlama2TrainInp…

### PR DESCRIPTION
**Pull request by sidhartha-roy**
This is related to the issue [SG-1153](https://protopia.atlassian.net/browse/SG-1153?atlOrigin=eyJpIjoiOGZlNGEwZWM3ZDg3NDc0MjljMDQwYTM3YTExMjgxMDkiLCJwIjoiaiJ9) and [SG-1132](https://protopia.atlassian.net/browse/SG-1132?atlOrigin=eyJpIjoiNjM0OGIyNjRiZmI3NDhmMWJhNTVhODI2MzczN2FhNDQiLCJwIjoiaiJ9) where the evaluation was failing because of the tensor shape mismatch. The solution was to update the tokenizer that includes the input tokens along with the output logits and ensure that the correct portions of the example texts are being used for evaluation.

[SG-1153]: https://protopia.atlassian.net/browse/SG-1153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SG-1132]: https://protopia.atlassian.net/browse/SG-1132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by KyleMylonakisProtopia — APPROVED** [review]

**Comment by Nan2018** [comment]
were you able to verify the fix with the previous checkpoint and score?

**Comment by sidhartha-roy** [comment]
> were you able to verify the fix with the previous checkpoint and score?

I tested it with OpenOrca after 5 epochs and I got 66% Hellaswag score, whereas previously I was getting 69%.

**Review by KyleMylonakisProtopia — CHANGES_REQUESTED** [review]
Blocking until we figure out discrepancy.
